### ISSUE TITLE
Not working - trying to help

### ISSUE THREAD

**Issue by EddyVerbruggen** [Marceel13/test-socialSharing#1]
Hey there,

How do you build this project? Using PhoneGap Build or a local build? Please elaborate and I may be able to assist.

**Comment by Marceel13** [comment]
I build it on local using the comand : phonegap local run android

**Comment by EddyVerbruggen** [comment]
- And does the console plugin work correctly (I saw you tried to configure that one as well)?
- Did you create the project with the CLI or did you throw some bits together yourself?

**Comment by Marceel13** [comment]
yes, all the logs outside and before of the plugin(social-sharing) functions are OK, but  the logs i try to write inside the plugin functions and after call those functions doesn't apear so i think i'm doing something wrong on the call. I created the project with the CLI.

**Comment by EddyVerbruggen** [comment]
Do you also see the logs on iOS? Because on Android console.log works also without this plugin.

A few things:
- Use `cordova.js`, [not `js/cordova.js`](https://github.com/Marceel13/test-socialSharing/blob/master/www/index.html#L14).
- Don't include a manual reference to [`SocialSharing.js`])https://github.com/Marceel13/test-socialSharing/blob/master/www/index.html#L17).
- Wait for `deviceready` to fire [before using a Cordova plugin](https://github.com/Marceel13/test-socialSharing/blob/master/www/index.html#L71).

**Comment by Marceel13** [comment]
Still not working, current commit have the changes. Though i don't really know how should the callback in deviceready work. If i have to wait to deviceready to use cordoba plugins, Do i have to assign the listeners to the buttons there with something like:

document.getElementById("fb").addEventListener("click",window.plugins.socialsharing.shareViaFacebook(three, null, null, function() {console.log('share ok');}, function(errormsg){alert(errormsg);}))   ?

**Comment by Marceel13** [comment]
i'm only using android, i don't have an iOS device.

**Comment by EddyVerbruggen** [comment]
[This won't work](https://github.com/Marceel13/test-socialSharing/blob/master/www/index.html#L69) as it's not waiting for `deviceready` to fire.

You could do it like this:

``` js
document.addEventListener("deviceready", function() {
    alert("device is ready, calling socialsharing");
    alert("plugins object: " + window.plugins);
    window.plugins.socialsharing.shareWithOptions(options, onSuccess, onError);
}, false);
```

**Comment by Marceel13** [comment]
I added what you said i see both alerts, on the second one  window.plugin is [object Object], i tried to add an other alert  alert("plugins : " + window.plugins.socialsharing); and i recive "plugins : undefined" , i guess thats a problem with the plugin instalation
